gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -449,13 +628,28 @@ private ManagedChannel createSingleChannel() throws IOException {
     } else {
       ChannelCredentials channelCredentials;
       try {
+        // Try and create credentials via DCA. See https://google.aip.dev/auth/4114.
         channelCredentials = createMtlsChannelCredentials();
       } catch (GeneralSecurityException e) {
         throw new IOException(e);
       }
       if (channelCredentials != null) {
+        // Create the channel using channel credentials created via DCA.
         builder = Grpc.newChannelBuilder(endpoint, channelCredentials);
       } else {
+        // Could not create channel credentials via DCA. In accordance with
+        // https://google.aip.dev/auth/4115, if credentials not available through
+        // DCA, try mTLS with credentials held by the S2A (Secure Session Agent).
+        if (useS2A) {
+          channelCredentials = createS2ASecuredChannelCredentials();
+        }
+        if (channelCredentials != null) {
+          // Create the channel using S2A-secured channel credentials.
+          builder = Grpc.newChannelBuilder(mtlsEndpoint, channelCredentials);
+        } else {
+          // Use default if we cannot initialize channel credentials via DCA or S2A.
+          builder = ManagedChannelBuilder.forAddress(serviceAddress, port);
+        }
         builder = ManagedChannelBuilder.forAddress(serviceAddress, port);
```

Comment:
I think we want to remove this last `builder = ManagedChannelBuilder.forAddress(serviceAddress, port);`  at L653. It looks like it would overwrite everything before it.

Reply:
Thanks for catching this (result of copy paste typo)!! Removed in https://github.com/googleapis/sdk-platform-java/pull/3548/commits/95e70d321b713a26812c07d7377b07fa4c22c2cb.